ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveSubQueryRemoveRule.java

```diff
@@ -230,33 +228,59 @@ private RexNode rewriteScalar(RelMetadataQuery mq, RexSubQuery e, Set<Correlatio
   private RexNode rewriteSomeAll(RexSubQuery e, Set<CorrelationId> variablesSet,
       HiveSubQRemoveRelBuilder builder) {
     final SqlQuantifyOperator op = (SqlQuantifyOperator) e.op;
-    assert(op == SqlStdOperatorTable.SOME_GE
-        || op == SqlStdOperatorTable.SOME_LE
-        || op == SqlStdOperatorTable.SOME_LT
-        || op == SqlStdOperatorTable.SOME_GT);
-    builder.push(e.rel)
-        .aggregate(builder.groupKey(),
-            op.comparisonKind == SqlKind.GREATER_THAN
-                || op.comparisonKind == SqlKind.GREATER_THAN_OR_EQUAL
-                ? builder.min("m", builder.field(0))
-                : builder.max("m", builder.field(0)),
-            builder.count(false, "c"),
-            builder.count(false, "d", builder.field(0)))
-        .as("q")
-        .join(JoinRelType.INNER);
-    return builder.call(SqlStdOperatorTable.CASE,
-        builder.call(SqlStdOperatorTable.EQUALS,
-            builder.field("q", "c"), builder.literal(0)),
-        builder.literal(false),
-        builder.call(SqlStdOperatorTable.IS_TRUE,
-            builder.call(RelOptUtil.op(op.comparisonKind, null),
-                e.operands.get(0), builder.field("q", "m"))),
-        builder.literal(true),
-        builder.call(SqlStdOperatorTable.GREATER_THAN,
-            builder.field("q", "c"), builder.field("q", "d")),
-        e.rel.getCluster().getRexBuilder().makeNullLiteral(SqlTypeName.BOOLEAN),
-        builder.call(RelOptUtil.op(op.comparisonKind, null),
-            e.operands.get(0), builder.field("q", "m")));
+    assert (op == SqlStdOperatorTable.SOME_GE || op == SqlStdOperatorTable.SOME_LE
+        || op == SqlStdOperatorTable.SOME_LT || op == SqlStdOperatorTable.SOME_GT);
+    if (variablesSet.isEmpty()) {
```

Comment:
Can we add comments below with the main operators we are creating (or at least an intuition on the rewriting)?